src/storage/src/sink/kafka.rs

```diff
@@ -1103,9 +1112,15 @@ where
                     shutdown_flush.set(true);
                 }
 
-                // Indicate that the sink is closed to everyone else who
-                // might be tracking its write frontier.
-                s.write_frontier.borrow_mut().clear();
+                // NOTE: This is somewhat subtle, but we never downgrade our
+                // write frontier to the empty frontier when we're shutting
+                // down. We might be shutting down for any number of reasons,
+                // most of them probably not because our source is finished.
+                // Meaning in most cases it would be wrong to advance our write
+                // frontier to the empty frontier.
+                //
+                // This note is here because a previous version of the code
+                // _did_ downgrade to the empty frontier here.
```

Comment:
IIUC, this is safe because in the rare case where we _should_ be clearing out the write frontier, it will be cleared automatically when a lease expires?

Reply:
writer leases don't hold back any frontiers. But yes, the rest of the code should still make it such that we downgrade to the empty frontier when that is what we actually want to do.

Reply:
just because I forget (sorry!): can you point to a test that exercises actually downgrading to the empty frontier?

is this comment because in the case where we hit some sort of rdkafka error (common cause of shutdown), we'd expect to be restarted and re-run the sink?  If not, can you give an example of when we'd shut down (in this way) and why empty frontier would be incorrect behavior?

Reply:
Sorry, got some wires crossed there! I think I follow now.